### PR TITLE
Redis cache

### PR DESCRIPTION
**Pull request by phamann**
The in-memory cache has served us well, but as other consumers start to use the graph-ql endpoint (such as myFT) this is pushing the constraints of the memory on the standard heroku dynos, causing us to scale them horizontally.

This adds another Redis backed cache abstraction class which mirrors the interface of the in-memory one so we can easily swap in/out where objects get persisted too.

I've chosen the CAPI backend as a test and opened PR to get peoples thoughts.

### CODE REVIEW

File: server/lib/backend-adapters/index.js [hunk 1]
```diff
@@ -12,11 +12,13 @@ import Myft from './myft';
 import TodaysTopics from './todays-topics';
 import Bertha from './bertha';
 
-import Cache from '../cache';
+import MemCache from '../cache';
```
Comment (ironsidevsquincy) [inline server/lib/backend-adapters/index.js]: filename -> `mem-cache.js`

File: server/lib/redis.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const redisClient = redis.createClient({
+	port: redisUrl.port,
+	host: redisUrl.hostname,
+	enable_offline_queue: false // fail fast when redis is down
+});
+
+redisClient.auth(redisUrl.auth.split(':')[1]);
+
+redisClient.on('error', err => {
+	console.log('Redis Error', err);
```
Comment (ironsidevsquincy) [inline server/lib/redis.js]: `logger.error`

**Comment by ironsidevsquincy** [comment]
This is now ready

@phamann, can you give it the once over again (not much has changed)

**Comment by phamann** [comment]
👍 LGTM

**Pull request merged**